scripts/validate/shellcheck

```diff
@@ -0,0 +1,11 @@
+#!/usr/bin/env bash
+set -e
+
+# Maintain an array of files to shellcheck not the best solution but will do for the time being
+FILES=()
+FILES+=("contrib/completion/bash/docker")
+FILES+=("scripts/validate/shellcheck")
```

Comment:
We should but the intent of this initial PR is to introduce shellcheck validation. I don't want it to end up like my original PR on the Moby project where it started to be very difficult to review and merge.

The idea is once we have that integrated to the build, we start adding the rest of `scripts/`